module-bsp/devices/temperature/CT7117.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "CT7117.hpp"
+
+#include <array>
+
+namespace
+{
+    enum class Registers
+    {
+        Temp          = 0x00,
+        Config        = 0x01,
+        Low_Temp_Set  = 0x02,
+        High_Temp_Set = 0x03,
+        ID            = 0x07
+    };
+
+    enum class ConfigReg
+    {
+        OTS  = (1 << 15),
+        F1   = (1 << 12),
+        F0   = (1 << 11),
+        ALTM = (1 << 9),
+        SD   = (1 << 8),
+        EM   = (1 << 7),
+        RES1 = (1 << 6),
+        RES0 = (1 << 5),
+        TO   = (1 << 4),
+        PEC  = (1 << 3),
+        CR1  = (1 << 2),
+        CR0  = (1 << 1),
+        OS   = (1 << 0)
+    };
+
+    using namespace bsp::devices::temperature::CT7117;
+
+    std::array<std::uint8_t, 2U> to_bytes(const std::uint16_t value)
+    {
+        return {static_cast<std::uint8_t>(value & 0xFF00 >> 8U), static_cast<std::uint8_t>(value & 0xFF)};
+    }
+
+    std::uint16_t from_bytes(std::array<std::uint8_t, 2U> bytes)
+    {
+        return bytes[0] << 8U | bytes[1];
+    }
+
+    ssize_t write_register16(const std::uint8_t dev_id,
+                             drivers::DriverI2C &i2c,
+                             const Registers address,
+                             const std::uint16_t reg)
+    {
+        const drivers::I2CAddress addr = {
+            .deviceAddress = dev_id, .subAddress = static_cast<std::uint32_t>(address), .subAddressSize = 1};
+
+        return i2c.Write(addr, &to_bytes(reg)[0], sizeof(reg));
+    }
+
+    std::optional<std::uint16_t> read_register16(const std::uint8_t dev_id,
+                                                 drivers::DriverI2C &i2c,
+                                                 const Registers address)
+    {
+        const drivers::I2CAddress addr = {
+            .deviceAddress = dev_id, .subAddress = static_cast<std::uint32_t>(address), .subAddressSize = 1};
+        std::array<std::uint8_t, 2U> ret_value{};
+
+        if (const auto result = i2c.Read(addr, &ret_value[0], ret_value.size()); result == ret_value.size()) {
+            return from_bytes(ret_value);
+        }
+        return std::nullopt;
+    }
+
+    units::Temperature from_raw(const std::uint16_t raw)
+    {
+        if ((raw & 0x8000) != 0) { // sign bit present
+            const auto integer =
+                static_cast<std::uint16_t>(((~raw + 1) & 0x7FFF) >> 7); // remove sign bit and shift to lower byte
```

Comment:
code was copier only from another file but here could be introduced lambda instead comment